### PR TITLE
first commit - conditional parameter

### PR DESCRIPTION
**Pull request by mda100**
editing backend "edit" api routes 
- [x] make all parameters conditionals
- [ ] test try and except (currently it raises empty exception -> is the default error message enough or does exception do something different [more specific error or stop the function]?)
- [ ] postman to check the api calls

### CODE REVIEW

**Comment by mda100** [comment]
this branch is now for making the backend edit api routes optional for pieces of data they are editing
error handling for api routes is now issue #4

File: .project [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+<projectDescription>
+	<name>salt_server</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+	</buildSpec>
+	<natures>
+	</natures>
+</projectDescription>
```
Comment (royalsaltmerchant) [inline .project]: We don't need this file in GitHub

**Comment by mda100** [comment]
Screen Shot 2021-09-09 at 4 08 39 PM
Screen Shot 2021-09-09 at 4 09 00 PM
Screen Shot 2021-09-09 at 4 09 45 PM
Screen Shot 2021-09-09 at 4 10 00 PM
Screen Shot 2021-09-09 at 4 10 25 PM

**Comment by mda100** [comment]
@royalsaltmerchant - the world's smallest backend change is ready to merge! here's some screenshots demonstrating the optionality of those two attributes now

**Review by royalsaltmerchant — APPROVED** [review]

**Pull request merged**